File: go/armstrong-numbers/armstrong_numbers.go
// Package armstrong implements a utility method to check if a number is armstrong number
package armstrong

import (
	"math"
)

// IsNumber returns true if number is an armstrong number
func IsNumber(n int) bool {
	digits := []int{}
	for i := n; i > 0; i /= 10 {
		digits = append(digits, i%10)
	}

	p := len(digits)
	sum := 0
	for _, d := range digits {
		sum += int(math.Pow(float64(d), float64(p)))
	}
	return n == sum
}

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        var digits = new List<int>();
        for (var n = number; n > 0; n /= 10)
        {
            digits.Add(n % 10);
        }

        var p = digits.Count;
        var sum = digits.Select(d => Math.Pow(d, p)).Sum();
        return number == sum;
    }
}